roles/semaphore_deployment/tasks/main.yml

```
@@ -0,0 +1,65 @@
+--- # Set up Ansible and restricted deployment user for deployment via Semaphore
+
+- name: install ansible
+  apt:
+    name: ansible
+  become: yes
+
+- name: create deployment user
+  user:
+    name: "{{ deployment_user }}"
+    state: present
+    append: yes
+    shell: /bin/bash
+  become: yes
+
+- name: add deploy script
+  template:
+    src: deploy.j2
+    dest: "/home/{{ deployment_user }}/deploy"
+    owner: root
+    mode: 0700
+
+- name: add sudoers file
+  template:
+    src: sudoers.j2
+    dest: "/etc/sudoers.d/{{ deployment_user }}"
+
+- name: fetch ofn-install repo
+  git:
+    repo: https://github.com/openfoodfoundation/ofn-install.git
+    dest: "/home/{{ deployment_user }}/ofn-install"
+    version: master
+    force: yes
+  tags: fetch_ofn_install
+
+- name: install community modules
+  shell: "/home/{{ deployment_user }}/ofn-install/bin/setup"
+  tags: update_community_modules
+
+- name: create secure directory for keys
+  file:
+    path: "/home/{{ user }}/keys"
+    state: directory
+    owner: "{{ user }}"
+    mode: 0700
+
+- name: check certificate already exists
+  stat:
+    path: "/home/{{ user }}/keys/semaphore.pub"
+  register: semaphore_key
+
+- name: generate semaphore key pair
+  command: "ssh-keygen -f /home/{{ user }}/keys/semaphore -t rsa -b 4096 -C semaphore -N '' "
+  when: not semaphore_key.stat.exists
```

You can replace the previous command with this simple condition:
```suggestion
  args:
    creates: "/home/{{ user }}/keys/semaphore.pub"
```

---

Thanks, I hadn't seen that trick before. I can scrap the preceding task as well :heart: